File: simplex.java/src/main/java/kr/co/tmon/simplex/actions/AbstractParameterizedAction.java
package kr.co.tmon.simplex.actions;

import kr.co.tmon.simplex.channels.Channel;

public abstract class AbstractParameterizedAction<TParam, TResult> extends AbstractAction<TResult> implements IParameterizedAction<TParam, TResult>{
	
	public AbstractParameterizedAction() {
		defaultChannel = new Channel();
	}
}

File: Simplex.kotlin/src/main/kotlin/kr/co/tmon/simplex/actions/AbstractParameterizedAction.kt
package kr.co.tmon.simplex.actions

import io.reactivex.Observable

abstract class AbstractParameterizedAction<TParam, TResult>
    : AbstractAction<TResult>(), IParameterizedAction<TParam, TResult> {

    abstract override fun process(param: TParam?) : Observable<TResult>

    override fun process() : Observable<TResult> {
        return process(null)
    }
}